File: 2024/august/java/03.java
class Solution {
    public boolean canBeEqual(int[] target, int[] arr) {
    final int mx = Arrays.stream(arr).max().getAsInt();
    int [] fa = new int[mx+1];
    int [] fb = new int[mx+1];
    for (var ch : arr) fa[ch]++;
    for (var ch : target) {
        if (ch <= mx) fb[ch]++;
    }

    for (int i = 0; i <= mx; ++i) if (fa[i] != fb[i]) return false;
    return true;
}
}

File: 2024/august/javaScript/03.js
/**
 * @param {number[]} target
 * @param {number[]} arr
 * @return {boolean}
 */
var canBeEqual = function(target, arr) {
    const mx = arr.reduce((p , cur)=> cur > p ? cur : p , 0);
    const fa = Array.from({length : mx+1} , ()=> 0);
    const fb = Array.from({length : mx+1} , ()=>0);
    for (const ch of arr) fa[ch]++;
    for (const ch of target) if (ch <= mx) fb[ch]++;

    for (let i = 0; i <= mx; ++i) if (fa[i] != fb[i]) return false;
    return true;
};
